tests/test_coremanager.py

```diff
@@ -21,35 +26,99 @@ def test_deptree():
+            ),
+        "::deptree-child4:0": (
+            "child4.sv",
+            ),
+        "::deptree-child-a:0": (
+            # Files from filesets are always included before any
+            # files from generators.
+            # This is because generated files are often dependent on files
+            # that are not generated, and it convenient to be able to
+            # include them in the same core.
+            "child-a2.sv",
+            "generated-child-a.sv",
+            ),
+        "::deptree-root:0": (
+            "root-fs1-f1.sv",
+            "root-fs1-f2.sv",
+            "root-fs2-f1.sv",
+            "root-fs2-f2.sv",
+            ),
+        }
+
     # Check dependency tree
     deps = cm.get_depends(root_core.name, {})
     deps_names = [str(c) for c in deps]
-    deps_names_expected = [
-        "::deptree-child2:0",
-        "::deptree-child3:0",
-        "::deptree-child1:0",
-        "::deptree-root:0",
-    ]
-    assert deps_names == deps_names_expected
-
-    # Check files in dependency tree
-    files_expected = [
-        "child2-fs1-f1.sv",
-        "child2-fs1-f2.sv",
-        "child3-fs1-f1.sv",
-        "child3-fs1-f2.sv",
-        "child1-fs1-f1.sv",
-        "child1-fs1-f2.sv",
-        "root-fs1-f1.sv",
-        "root-fs1-f2.sv",
-        "root-fs2-f1.sv",
-        "root-fs2-f2.sv",
-    ]
-    files = []
-    for d in deps:
-        files += [f["name"] for f in d.get_files({})]
+    
+    all_core_names = set()
+    for child, parent in dependencies:
+        assert child in deps_names
+        assert parent in deps_names
+        if not (deps_names.index(child) < deps_names.index(parent)):
+            import pdb
+            pdb.set_trace()
+        all_core_names.add(child)
+        all_core_names.add(parent)
+    # Confirm that we don't have any extra or missing core names.
+    assert all_core_names == set(deps_names)
+    # Make sure there are no repeats in deps_names
+    assert len(all_core_names) == len(deps_names)
+
+    # Now work out what order we expect to get the filenames.
+    # The order of filenames within each core in deterministic.
+    # Each fileset in order. Followed by each generator in order.
+    # The order between the cores is taken the above `dep_names`.
+    expected_filenames = []
+    for dep_name in deps_names:
+        expected_filenames += list(expected_core_files[dep_name])
+
+    # Use Edalizer to get the files.
+    # This is necessary because we need to run generators.
+    work_root = os.path.join(temp_dir, 'work')
+    cache_root = os.path.join(temp_dir, 'cache')
+    os.mkdir(work_root)
+    os.mkdir(cache_root)
+    edalizer = Edalizer(
+        toplevel=root_core.name,
+        flags=flags,
+        cache_root=cache_root,
+        work_root=work_root,
+        core_manager=cm,
+        )
+    edalizer.run()
+    edalized_filenames = [os.path.basename(f['name']) for f in edalizer.edalize['files']]
+
+    assert expected_filenames == edalized_filenames
```

Comment:
```suggestion
    assert edalized_filenames == expected_filenames
```

to make pytest's diff view correct